src/Components/Ignitor/src/CancellableOperation.cs

```diff
@@ -5,37 +5,50 @@
 using System.Threading;
 using System.Threading.Tasks;
 
+#nullable enable
 namespace Ignitor
 {
     internal class CancellableOperation<TResult>
     {
-        public CancellableOperation(TimeSpan? timeout)
+        public CancellableOperation(TimeSpan? timeout, CancellationToken cancellationToken)
```

Comment:
The bulk of the interesting changes are localized to this type. Here's what happens:

1) We start a `PrepareForNextBatch`
2) In the meanwhile, `BlazorClient.Cancel` is invoked (the server may or may not be dead at this point)
3) `CancellableOperation` knows nothing about this and eventually times out.

This change associates the `BlazorClient.Cancel` with the operation and no-ops if the client has been cancelled